crates/metaslang/cst/src/cursor.rs

```diff
@@ -422,50 +412,42 @@ impl<T: KindTypes> Cursor<T> {
     }
 }
 
-/// A [`Cursor`] that also keeps track of the labels of the nodes it visits.
-pub struct CursorWithEdges<T: KindTypes> {
+pub struct CursorIterator<T: KindTypes> {
     cursor: Cursor<T>,
 }
 
-impl<T: KindTypes> CursorWithEdges<T> {
-    pub fn without_edges(self) -> Cursor<T> {
-        self.cursor
-    }
-}
-
-impl<T: KindTypes> Iterator for CursorWithEdges<T> {
+impl<T: KindTypes> Iterator for CursorIterator<T> {
     type Item = Edge<T>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let label = self.cursor.label();
+        if self.cursor.is_completed() {
+            return None;
+        }
 
-        self.cursor.next().map(|node| Self::Item { label, node })
-    }
-}
+        let current = Edge {
+            label: self.cursor.label(),
+            node: self.cursor.node().clone(),
```

Comment:
The `clone()` here is redundant as the cursor's node is already cloned inside `node()`.

Reply:
Fixed!